src/upnp/quirks.cc

```diff
@@ -55,7 +55,7 @@ long Quirks::checkFlags(long flags) const
 
 bool Quirks::hasFlag(long flag) const
 {
-    return pClientInfo && (pClientInfo->flags & flag) == flag;
+    return pClientInfo && (static_cast<long>(pClientInfo->flags) & flag) == flag;
```

Comment:
There is another such line in `checkFlags`. But I prefer to replace the long by `QuirkFlags`, which requires changes in `ConfigOption` as well.
As I'm already in the middle of these changes, you can remove that change.

Reply:
Removed.